homeassistant/components/asuswrt/__init__.py

```diff
@@ -1,18 +1,44 @@
 """Support for ASUSWRT devices."""
 
+import logging
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import EVENT_HOMEASSISTANT_STOP, Platform
 from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
 
+from .bridge import migrate_legacy_protocols
 from .const import DATA_ASUSWRT, DOMAIN
 from .router import AsusWrtRouter
 
+MIGRATE_HTTP_FLOW_VERSION = 2
 PLATFORMS = [Platform.DEVICE_TRACKER, Platform.SENSOR]
 
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", entry.version)
+
+    if entry.version < MIGRATE_HTTP_FLOW_VERSION:
+        if (new_conf := await migrate_legacy_protocols(hass, dict(entry.data))) is None:
+            return True
+
+        entry.version = MIGRATE_HTTP_FLOW_VERSION
+        if new_conf:
+            hass.config_entries.async_update_entry(entry, data=new_conf)
+
+        _LOGGER.debug("Migration to version %s successful", entry.version)
+
+    return True
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up AsusWrt platform."""
 
+    if entry.version < MIGRATE_HTTP_FLOW_VERSION:
```

Comment:
If someone uses the ssh key, he can't use the integration anymore. Why?

Reply:
Ok, this is the main open point for this PR. The problem is that `ssh` can for sure an be used, but is not possible to automatic migrate to `http` because we don't have a password. On initial review the idea was to launch a `reauth` in this condition, but I'm not totally sure that this is the right approach. Maybe we could lust leave the integration run with current configuration and create a [Reapair](https://developers.home-assistant.io/docs/core/platform/repairs) to notify the user about new protocol option?